source/CMakeLists.txt

```diff
@@ -24,6 +24,84 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
 
+set(GRAMMAR_PROCESSING_SCRIPT ${spirv-tools_SOURCE_DIR}/utils/generate_grammar_tables.py)
+
+# macro() definitions are used in the following because we need to append .inc
+# file paths into some global lists (*_CPP_DEPENDS). And those global lists are
+# later used by set_source_files_properties() calls.
+# function() definitions are not suitable because they create new scopes.
+macro(spvtools_core_tables VERSION)
+  set(GRAMMAR_JSON_FILE ${CMAKE_CURRENT_SOURCE_DIR}/spirv-${VERSION}.core.grammar.json)
```

Comment:
I believe we are defining variables in the `source/` directory names here. So, should not be a great problem unless somebody include `source/CMakeLists.txt` directory.